README.md

```diff
@@ -1,83 +1,83 @@
 
-As the name suggests, the purpose of the *fallback* is to allow a fallback when, for example,
+As the name suggests, the purpose of the *alternate* is to allow a alternate when, for example,
 the desired upstreams became unavailable.
 
 ## Syntax
 
 ```
 {
-    fallback [original] RCODE PROXY_PARAMS
+    alternate [original] RCODE FORWARD_PARAMS
 }
 ```
 
-* **original** is optional flag. If it is set then fallback uses original request instead of potentially changed by other plugins
+* **original** is optional flag. If it is set then alternate uses original request instead of potentially changed by other plugins
 * **RCODE** is the string representation of the error response code. The complete list of valid rcode strings are defined as `RcodeToString` in <https://github.com/miekg/dns/blob/master/msg.go>, examples of which are `SERVFAIL`, `NXDOMAIN` and `REFUSED`.
-* **PROXY_PARAMS** accepts the same parameters as the *proxy* plugin
-<https://coredns.io/plugins/proxy>.
+* **FORWARD_PARAMS** accepts the same parameters as the *forward* plugin
+<https://coredns.io/plugins/forward>.
 
 ## Examples
 
-### Fallback to local DNS server
+### Alternate to local DNS server
 
-The following specifies that all requests are proxied to 8.8.8.8. If the response is `NXDOMAIN`, *fallback* will proxy the request to 192.168.1.1:53, and reply to client accordingly.
+The following specifies that all requests are forwarded to 8.8.8.8. If the response is `NXDOMAIN`, *alternate* will forward the request to 192.168.1.1:53, and reply to client accordingly.
 
 ```
 . {
-	proxy . 8.8.8.8
-	fallback NXDOMAIN . 192.168.1.1:53
+	forward . 8.8.8.8
+	alternate NXDOMAIN . 192.168.1.1:53
 	log
 }
 
 ```
-### Fallback with original request used
+### Alternate with original request used
 
-The following specify that `original` query will be proxied to 192.168.1.1:53 if 8.8.8.8 response is `NXDOMAIN`. `original` means no changes from next plugins on request. With no `original` flag fallback will proxy request with EDNS0 option (set by rewrite).
+The following specify that `original` query will be forwarded to 192.168.1.1:53 if 8.8.8.8 response is `NXDOMAIN`. `original` means no changes from next plugins on request. With no `original` flag alternate will forward request with EDNS0 option (set by rewrite).
 
 ```
 . {
-	proxy . 8.8.8.8
+	forward . 8.8.8.8
     rewrite edns0 local set 0xffee 0x61626364
-	fallback original NXDOMAIN . 192.168.1.1:53
+	alternate original NXDOMAIN . 192.168.1.1:53
 	log
 }
 
 ```
 
-### Multiple fallbacks
+### Multiple alternates
 
-Multiple fallbacks can be specified, as long as they serve unique error responses.
+Multiple alternates can be specified, as long as they serve unique error responses.
 
 ```
 . {
-    proxy . 8.8.8.8
-    fallback NXDOMAIN . 192.168.1.1:53
-    fallback REFUSED . 192.168.100.1:53
-    fallback original SERVFAIL . 192.168.100.1:53
+    forward . 8.8.8.8
+    alternate NXDOMAIN . 192.168.1.1:53
```

Comment:
current plugin/forward does not allow to have several directive FORWARD in the same stanza.
I am not sure why, but did you check there will be no issue to use on FORWARD in each of the Rules of Alternate ?

Reply:
Yes. I have tried and it worked fine.

Reply:
The only reason I see they don't support multiple plugins is because you can have only one plugin registered per stanza and multiple can be only "emulated" which requires additional work to do. But forward plugin is made with OOP approach and I don't see any global states there.
As we have our own "stanza", so we can have as many plugins as we want and they don't intersect.

Reply:
ok. Thanks!